controllers/managedocs_controller.go

```diff
@@ -550,6 +574,20 @@ func (r *ManagedOCSReconciler) reconcilePrometheus() error {
 		return err
 	}
 
+	_, err = ctrl.CreateOrUpdate(r.ctx, r.Client, r.dmsRule, func() error {
+		if err := r.own(r.dmsRule); err != nil {
+			return err
+		}
+
+		desired := templates.DMSPrometheusRuleTemplate.DeepCopy()
+		r.dmsRule.Spec = desired.Spec
+
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+
```

Comment:
I would suggest to keep a single reconciled entity per reconcile func.
This can be moved to a `reconcileDMSPrometheusRule` func (called from reconcilePhases)

Reply:
Please explain the benefit of moving keeping a single reconciled entity per reconcile func.

Reply:
The current code division (`reconcilePhases` calling `reconcileXXXX` where XXXX is a specific resource) creates an expectation for a reader/contributor when learning the codebase. 

Creating an exception in this case without a specific reason (unless I am missing something that forces us to put these two `CreateOrUpdate` invocations into the same func), and where the alternative (a `reconcileDMSPrometheusRule` func) has no visible cons (again, unless I am missing something), just introduce noise into the current code structure. 

Now. One can argue that the original code division is not convenient or does not provide value. My answer to that is that without pointing to any specific flaw, any other solution is just a subjective opinion. And as a matter of fact, if we are going to break this structure, let's not do it arbitrarily, which means we should do it for all objects and unfold everything into `reconcilePhases` (I am not really advocating for this, just pointing out that this exception is arbitrary)